lib/devices/light.js

```diff
@@ -27,103 +31,144 @@ module.exports = function(device, api, capabilities) {
-      console.log('created listener for - ' + device.capabilities[i]);
-      let listener = async (value) => {
-        onStateChange(device.capabilities[i], value, device);
-      };
-
-      device.makeCapabilityInstance(device.capabilities[i], listener);
-    }
-  }
 
-  async function onStateChange(capability, value, device) {
-    console.log('State Change - ' + device.name + ' - ' + capability + ' - ' + value);
-    const light = homekitAccessory.getService(Service.Lightbulb);
-
-    if (capability === 'onoff') {
-      light.getCharacteristic(Characteristic.On).updateValue(value);
-    } else if (capability === 'dim') {
-      light.getCharacteristic(Characteristic.Brightness).updateValue(value * 100);
-    } else if (capability === 'light_saturation') {
-      light.getCharacteristic(Characteristic.Saturation).updateValue(value * 100);
-    } else if (capability === 'light_temperature') {
-      light.getCharacteristic(Characteristic.ColorTemperature).updateValue(map(0, 1, 140, 500, value));
-    } else if (capability === 'light_hue') {
-      light.getCharacteristic(Characteristic.Hue).updateValue(value * 360);
-    }
+const registerCapabilityListener = ({ device, capability, service }) => {
+  console.log('created listener for - ' + capability);
+
+  const listener = (value) => {
+    onStateChange({ service, capability, value, device });
+  };
+
+  // For Homey device we want to use full capability name
+  device.makeCapabilityInstance(capability, listener);
+}
+
+const onStateChange = ({ service, capability, value, device }) => {
+  console.log('State Change - ' + device.name + ' - ' + capability + ' - ' + value);
+  const { capabilityBaseName } = getCapabilityNameSegments(capability)
+
+  if (capabilityBaseName === 'onoff') {
+    service.getCharacteristic(Characteristic.On).updateValue(value);
+  } else if (capabilityBaseName === 'dim') {
+    service.getCharacteristic(Characteristic.Brightness).updateValue(value * 100);
+  } else if (capabilityBaseName === 'light_saturation') {
+    service.getCharacteristic(Characteristic.Saturation).updateValue(value * 100);
+  } else if (capabilityBaseName === 'light_temperature') {
+    service.getCharacteristic(Characteristic.ColorTemperature).updateValue(map(0, 1, 140, 500, value));
+  } else if (capabilityBaseName === 'light_hue') {
+    service.getCharacteristic(Characteristic.Hue).updateValue(value * 360);
   }
+}
 
-  // Return device to app.js
-  return homekitAccessory
+const isLightCapability = (capability) => {
+  const { capabilityBaseName } = getCapabilityNameSegments(capability)
+  return ['onoff', 'dim', 'light_saturation', 'light_temperature', 'light_hue'].includes(capabilityBaseName);
 }
+
+const getServiceBuildArguments = ({ device, groupName, groups, group }) => {
+  const isSingleService = groups.length === 1;
+  const serviceName = isSingleService ? device.name : getServiceName({ device, group })
+
+  return [Service.Lightbulb, serviceName, groupName];
```

Comment:
@Sinled I do have a question about this: this array is used for arguments to `addService()`. The previous code only pass two arguments to that method, but you're passing an extra (`groupName`). I can't find any specific documentation on whether `addService` actually takes a third argument, but is it necessary?

Because I don't use my Homey anymore, I asked @nhassink to test with his setup, and he found that all sockets and lights that he has were re-added to HomeKit. Could passing this third argument be the cause of that?